src/client.rs

```diff
@@ -172,18 +125,139 @@ impl Client {
+        async fn verify(&self, _collection: &Collection) -> Result<(), SignatureError> {
+            Ok(())
+        }
+    }
+
+    #[async_trait]
+    impl Verification for VerifierWithInvalidSignatureError {
+        async fn verify(&self, _collection: &Collection) -> Result<(), SignatureError> {
+            return Err(SignatureError::InvalidSignature{name: "invalid signature error".to_owned()});
+        }
+    }
+
+    #[tokio::test]
+    #[with_mock_server]
+    async fn test_get_fails_if_verification_fails_with_verification_error() {
+        let get_changeset_mock = mock(GET, "/buckets/main/collections/url-classifier-skip-urls/changeset")
+        .expect_query_param("_expected", "0")
+        .return_status(200)
+        .return_header("Content-Type", "application/json")
+        .return_body(
+            r#"{
+                "metadata": {},
+                "changes": [],
+                "timestamp": 0
+            }"#
+        ).create();
+
+        let actual_result = Client::create_with_server_collection("http://localhost:5000", "url-classifier-skip-urls", Some(Box::new(VerifierWithVerificatonError{}))).get(0).await;
+        let expected_result = Err(ClientError::VerificationError{name: "verification error".to_owned()});
+        assert_eq!(1, get_changeset_mock.times_called());
+        assert_eq!(expected_result, actual_result);
     }
 
+    #[tokio::test]
+    #[with_mock_server]
+    async fn test_get_passes_if_verification_passes() {
+        let expected_version: u64 = 10;
+
+        let get_changeset_mock = mock(GET, "/buckets/main/collections/url-classifier-skip-urls/changeset")
+        .expect_query_param("_expected", &expected_version.to_string())
+        .return_status(200)
+        .return_header("Content-Type", "application/json")
+        .return_body(
+            r#"{
+                "metadata": {
+                    "data": "test"
+                },
+                "changes": [{
+                    "id": 1,
+                    "last_modified": 100
+                }],
+                "timestamp": 0
+            }"#
+        ).create();
+
+        let actual_result = Client::create_with_server_collection("http://localhost:5000", "url-classifier-skip-urls", Some(Box::new(VerifierWithNoError{}))).get(expected_version).await;
+        assert_eq!(1 , get_changeset_mock.times_called());
+
+        match actual_result {
+            Ok(collection) => {
+                assert_eq!(
+                    Collection {
+                        bid: "main".to_owned(),
+                        cid: "url-classifier-skip-urls".to_owned(),
+                        metadata: json!({
+                            "data": "test".to_owned()
+                        }),
+                        records: vec![
+                            json!({
+                                "id": 1,
+                                "last_modified": 100
+                            })
+                        ],
```

Comment:
`.get()` is supposed to return only records right?

Reply:
After reading the filtering code in #16 I can see that you return a `Collection` object. IMO `.get()` should really return records only